packages/legend-extension-dsl-data-space/src/graph-manager/action/analytics/DataSpaceAnalysis.ts

```diff
@@ -173,7 +177,7 @@ export class DataSpaceExecutableAnalysisResult {
 
   title!: string;
   description?: string | undefined;
-  executable!: string;
+  executable?: string;
```

Comment:
I'm not super familiar with data spaces yet. What's the rationale behind changing this parameter from required to optional?

Reply:
For template query, this field is optional. 

e.g. 
`  "executables": [
    {
      "description": "Some more exec description",
      "info": {
        "_type": "templateExecutableInfo",
        "executionContextKey": "dummyContext",
        "query": "|domain::COVIDData.all()->project([x: domain::COVIDData[1]|$x.cases], ['Cases'])"
      },
      "result": {
        "_type": "tds",
        "columns": [
          {
            "name": "Cases",
            "relationalType": "INTEGER",
            "type": "Float"
          }
        ]
      },
      "title": "this is template"
    }
  ],`

check public void testAnalyticsForDataSpaceTemplateWithExecutables() throws Exception in 
https://github.com/YannanGao-gs/legend-engine/blob/4eb1738bd03659fd7ee536ea655773b38020bb43/legend-engine-xts-data-space/legend-engine-xt-data-space-generation/src/test/java/org/finos/legend/engine/generation/TestDataSpaceAnalyticsArtifactGenerationExtension.java